src/senaite/patient/content/patient.py

```diff
@@ -247,6 +257,20 @@ def validate_patient_id(data):
         if patient:
             raise Invalid(_("Patient ID must be unique"))
 
+    @invariant
+    def validate_patient_email_report(data):
+        """Checks if an email is set
+        """
+        value = data.email_report
+        if not value:
+            return
+
+        # https://community.plone.org/t/dexterity-unique-field-validation
+        context = getattr(data, "__context__", None)
+        if context is not None:
+            if not context.getEmail():
+                raise Invalid(_("Please set a valid email Address first"))
```

Comment:
Maybe the email was empty and is being set in current request, along with the assignment of "EmailReport". In such case, the error will arise. Likewise, if the email is emptied in this same request, the error won't arise, but "EmailReport" will be set to True.